### ISSUE TITLE
manual_swap in const context false positive

### ISSUE THREAD

**Issue by pacak** [rust-lang/rust-clippy#9864]
### Summary

clippy suggests to use std::mem::swap in const context on stable.

### Lint Name

clippy::manual_swap

### Reproducer

I tried this code:

```rust
pub const fn foo(mut u: u32) -> u32 {
    let mut v = 10;

    let temp = u;
    u = v;
    v = temp;
    u + v
}
```

I saw this happen:

```
warning: this looks like you are swapping `u` and `v` manually
   --> src/lib.rs:264:5
    |
264 | /     let temp = u;
265 | |     u = v;
266 | |     v = temp;
    | |____________^ help: try: `std::mem::swap(&mut u, &mut v)`
    |
    = note: `#[warn(clippy::manual_swap)]` on by default
    = note: or maybe you should use `std::mem::replace`?
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_swap
```
Suggested code fails with "`std::mem::swap` is not yet stable as a const fn"

### Version

```text
rustc 1.65.0 (897e37553 2022-11-02)
binary: rustc
commit-hash: 897e37553bba8b42751c67658967889d11ecd120
commit-date: 2022-11-02
host: x86_64-unknown-linux-gnu
release: 1.65.0
LLVM version: 15.0.0
```
